lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -0,0 +1,183 @@
+
+    // Increase the current time by the specified time.
+    this.#now += time;
+
+    // Execute all timers whose runAt time is less than or equal to the current time.
+    let timer = this.#timers.peek();
+    while (timer && timer.runAt <= this.#now) {
+      // Execute the timer's callback function with the specified arguments.
+      timer.callback(...timer.args);
+
+      // Remove the timer from the timer queue.
+      this.#timers.shift();
+
+      // If the timer is an interval timer, update its runAt time and re-insert it into the timer queue.
+      if (timer.interval) {
+        timer.runAt += timer.interval;
+        this.#timers.insert(timer);
+        return;
+      }
+
+      // Get the next timer in the timer queue.
+      timer = this.#timers.peek();
+    }
+
+  }
+
+  enable() {
+    // if (this.isEnabled) {
+    //   throw new Error('fakeTimers is already enabled!');
+    // }
+    this.#now = DateNow();
+    this.#isEnabled = true;
+
+    this.#realSetTimeout = globalThis.setTimeout;
+    this.#realClearTimeout = globalThis.clearTimeout;
+    this.#realSetInterval = globalThis.setInterval;
+    this.#realClearInterval = globalThis.clearInterval;
+
+    this.#realTimersSetTimeout = nodeTimers.setTimeout;
+    this.#realTimersClearTimeout = nodeTimers.clearTimeout;
+    this.#realTimersSetInterval = nodeTimers.setInterval;
+    this.#realTimersClearInterval = nodeTimers.clearInterval;
+
+    this.#realPromisifiedSetTimeout = nodeTimersPromises.setTimeout;
+    this.#realPromisifiedSetInterval = nodeTimersPromises.setInterval;
+
+    globalThis.setTimeout = this.#setTimeout;
+    globalThis.clearTimeout = this.#clearTimeout;
+    globalThis.setInterval = this.#setInterval;
+    globalThis.clearInterval = this.#clearInterval;
+
+    nodeTimers.setTimeout = this.#setTimeout;
+    nodeTimers.clearTimeout = this.#clearTimeout;
+    nodeTimers.setInterval = this.#setInterval;
+    nodeTimers.clearInterval = this.#clearInterval;
+
+    nodeTimersPromises.setTimeout = (ms) => new Promise(
+      (resolve) => {
+        const id = this.#setTimeout(() => resolve(id), ms);
+      },
+    );
+
+    nodeTimersPromises.setInterval = FunctionPrototypeBind(this.#setIntervalPromisified, this);
+  }
+
+  reset() {
+    this.#isEnabled = false;
+
+    // Restore the real timer functions
+    globalThis.setTimeout = this.#realSetTimeout;
+    globalThis.clearTimeout = this.#realClearTimeout;
+    globalThis.setInterval = this.#realSetInterval;
+    globalThis.clearInterval = this.#realClearInterval;
+
+    nodeTimers.setTimeout = this.#realTimersSetTimeout;
+    nodeTimers.clearTimeout = this.#realTimersClearTimeout;
+    nodeTimers.setInterval = this.#realTimersSetInterval;
+    nodeTimers.clearInterval = this.#realTimersClearInterval;
+
+    nodeTimersPromises.setTimeout = this.#realPromisifiedSetTimeout;
+    nodeTimersPromises.setInterval = this.#realPromisifiedSetInterval;
+
+    let timer = this.#timers.peek();
+    while (timer) {
```

Comment:
We can avoid code repetition here
```suggestion
    while (timer = this.#timers.peek()) {
```

Reply:
According to the eslint config Node.js is using, I cannot do that. Got "  178:12  error  Expected a conditional expression and instead saw an assignment  no-cond-assign"

and did this: `while (timer = this.#timers.peek()) {`